oh-my-zsh.sh

```diff
@@ -59,9 +59,54 @@ if [ -z "$ZSH_COMPDUMP" ]; then
   ZSH_COMPDUMP="${ZDOTDIR:-${HOME}}/.zcompdump-${SHORT_HOST}-${ZSH_VERSION}"
 fi
 
-# Load and run compinit
-autoload -U compinit
-compinit -i -d "${ZSH_COMPDUMP}"
+# If one or more completion-dependent directories have insecure ownership or
+# permissions, the low-level compaudit() function fails with output resembling:
+#
+#     There are insecure directories:
+#     /usr/share/zsh/site-functions
+#     /usr/share/zsh/5.0.6/functions
+#     /usr/share/zsh
+#     /usr/share/zsh/5.0.6
+#
+# Silently detect and, in a few well-known cases, offer to fix this issue.
+if ! compaudit &>/dev/null; then
+  # Print this list of insecure directories.
+  compaudit
+  print
+
+  # If this is Cygwin *and* both standard input and output are connected to an
+  # interactive terminal, offer to fix this. Cygwin has a long-standing bug in
```

Comment:
> There is this one in the mailing list: Zsh completion problem but it looks like it was solved.

**The plot thickens.** For sanity, Babun forces use of the most recently released of Cygwin (2.0.4, as of this thick-headed writing). I'm happy to personally confirm that this issue is no longer solved, probably due to an undetected regression. The version of `zsh` cited in that **2011** mailing list thread (4.3.11) is obscenely obsolete. The long road to `zsh` 5.0.0 appears to have reintroduced this issue... *in stealth mode*.

But don't take my greasy word for it:

* Similar [Stackoverflow question](https://superuser.com/questions/489831/zsh-autocompletion-not-working-on-cygwin). **2012** >= 2011.
* Similar [`oh-my-cygwin` issue](https://github.com/haithembelhaj/oh-my-cygwin/issues/1). **2013**  >= 2011.
* Similar [Stackoverflow question](https://stackoverflow.com/questions/13762280/zsh-compinit-insecure-directories/22753363#22753363). **2014** >= 2011.
* Similar [`zsh` mailing list issue](http://www.zsh.org/mla/users/2014/msg00820.html). **2014** >= 2011.

Alternately, peruse user comments for the issues listed above in the pull request description. By inspection, this issue *clearly* remains unresolved.

> Also, in this blog post the author says this one-liner didn't fix it for him because, basically, the Windows ACLs weren't properly set up for the user that run cygwin.

Fair enough! Where there exists a discrepancy between the account with which Babun and/or Cygwin was installed (typically, Administrator) and the account attempting to use Babun and/or Cygwin, the provided solution may be insufficient. **But that's O.K.**

In such case, `oh-my-zsh` startup gracefully detects the failure to correct this issue and emits the non-fatal warning `"[oh-my-zsh] Unable to fix."`. This is *still* demonstrably superior to the current approach, in that the user is at least informed of the underlying issue in a human-readable manner.

This pull request is *not* intended to perfectly, completely, and finally address this issue for all users under all contexts. It *is* intended to address this issue for most Cygwin users under the simplistic assumption of parity between the account having installed and the account attempting to use Babun and/or Cygwin. This is generally the case, except where it isn't.

> So it looks to me that this PR will need a little more testing. @leycec can you direct some babun users this way? Let's see if they can confirm this solves compinit problems in a fresh installation of babun/cygwin.

That's... a good idea. By listing several Babun issues above in the pull request description, Github implicitly referenced this pull request in all such issues. Users who have commented on and hence implicitly subscribed to those issues *should* now theoretically be informed.